### DISCUSSION TITLE
Welcome to ACE Discussions!

### DISCUSSION THREAD

**Discussion by LtRipley36706** [ACEmulator/ACE#3384] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sLackeR-hub** [comment]
Hi I'm Slacker (Patrick) and I am an alcoholic..... Wait wrong discussion. 
I like re-creating quests that are missing from ACE currently, so if there are any quests missing that may be your favorite please let me know.

DT(Asheron'sCall)4lyfe
![IMG_0452](https://user-images.githubusercontent.com/70003173/101554072-8f210a80-397b-11eb-8798-413d7fae338f.jpg)

**Comment by jsprenkle** [comment]
Hello, I have a question about an error while compiling ACViewer. It gives an error from something called "monogame builder." Do I need to add the package from nuget?
Jay (AKA  Uzik from Solclaim)

**Reply by gmriggs** [reply]
Hey Jay,

This thread might contain some useful info:

https://community.monogame.net/t/monogame-content-builder-targets-error-message/16275/4

Maybe try doing step 1 mentioned in the last post

I don't have an install of a fresh Windows environment to test on currently, but I thought everything in the ACViewer project was updated to the latest 3.8 pipeline. Maybe the shader build task is still using an older process. If that is the case, I can add it to the todo list of things to look into.

**Reply by gmriggs** [reply]
Looking at ACViewer.csproj, it seems like nuget should automatically be pulling in the latest 3.8 version of the MonoGame.Content.Builder.Task. Make sure nuget is running and pulling everything in first!

The ACE submodule also needs to be pulled in via one of the scripts in the build folder, in case you run into any errors related to that

**Reply by jsprenkle** [reply]
> ACE submodule

At first I only pulled down the ACViewer code, but then found the dependencies to the ACE projects. Pulled them down too and they built without issue. It might be the new version of Visual Studio has done something to break the build. I looked for a bit for the 2019 version, but Microsoft doesn't want us to be able to find it easily!

Thanks

**Reply by gmriggs** [reply]
I've been using VS2022 to build ACViewer. Are you using the full version, or Community Edition?

**Reply by jsprenkle** [reply]
Community edition. I uninstalled the monogame nuget package (version 3.8) and ran the executable installer for version 3.7. I'm now able to compile and debug the app!  :)
They've made a lot of changes in the years since I've upgraded

**Reply by gmriggs** [reply]
Glad to hear it!

I will add it to my todo list to find out what is still needed from the 3.7 pipeline, as I thought everything was upgraded to 3.8 already

Thanks for the info, it is much appreciated

**Reply by jsprenkle** [reply]
I should have, but didn't document the steps I took to get it working. I do recall:

- The post build event copy statement did not work. It appeared to be related to deployment so I removed it.
- When running in debug mode loading UI resources fails because the working directory is different. Changing it in the debug launch properties will not fix the issue (limitation of Visual Studio). You can't add the resources directory as copied content when building (limitation of Visual Studio). I finally had to copy all resources manually to the debug working directory ( bin/debug/something_I_forget/ )

I've given up on trying to contribute to projects. Even when they solicit them they aren't really welcome or won't make any effort. If the project is active and contributions will be welcomed I'd be willing to give it another try.
